### PR TITLE
Add test retry plugin for GE plugin versions supporting safe mode

### PR DESCRIPTION
**Pull request by marcphilipp**

### CODE REVIEW

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/BuildScanPluginSmokeTest.groovy
Comment (marcphilipp) [inline subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/BuildScanPluginSmokeTest.groovy:1]: `AbstractSmokeTest.TestedVersions.teamCityGradlePluginRef` etc. need to be adjusted after the following PRs are merged:
- https://github.com/etiennestuder/teamcity-build-scan-plugin/pull/287
- https://github.com/gradle/gradle-enterprise-bamboo-plugin/pull/55
- https://github.com/jenkinsci/gradle-plugin/pull/324

**Comment by marcphilipp** [comment]
@bot-gradle test this

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/BuildScanPluginSmokeTest.groovy [hunk 1]
```diff
@@ -147,6 +148,8 @@ class BuildScanPluginSmokeTest extends AbstractSmokeTest {
     private static final VersionNumber FIRST_VERSION_SUPPORTING_GRADLE_8_CONFIGURATION_CACHE = VersionNumber.parse("3.12")
     private static final VersionNumber FIRST_VERSION_SUPPORTING_ISOLATED_PROJECTS = VersionNumber.parse("3.15")
     private static final VersionNumber FIRST_VERSION_CALLING_BUILD_PATH = VersionNumber.parse("3.13.1")
+    private static final VersionNumber FIRST_VERSION_BUNDLING_TEST_RETRY_PLUGIN = VersionNumber.parse("3.12")
+    private static final VersionNumber FIRST_VERSION_SUPPORTING_SAFE_MODE = VersionNumber.parse("3.15")
```
Comment (marcphilipp) [inline subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/BuildScanPluginSmokeTest.groovy:152]: 💬 3.15 is not yet covered here. I published a version locally and tested it using this test so it should continue to work when 3.15 is added.

**Review by wolfs — APPROVED** [review]

**Comment by marcphilipp** [comment]
@bot-gradle test and merge

**Pull request merged**